pkg/workflow/hooks/data_passing.go

```diff
@@ -66,13 +56,6 @@ func Output(ctx wfContext.Context, taskValue *value.Value, step v1beta1.Workflow
 			if err := json.Unmarshal(js, &o); err != nil {
 				return err
 			}
-			ready, err := value.NewValue(`true`, nil, "")
-			if err != nil {
-				return err
-			}
-			if err := ctx.SetVar(ready, ReadyComponent, o.Name); err != nil {
-				return err
-			}
```

Comment:
why don't we need this here?

Reply:
I pass all the step status as a map to the `Pending` so we can get the status from the map directly, there's no need that we store another copy of ready step status in ctx.